test/error.js

```diff
@@ -1,19 +1,26 @@
 'use strict';
 
-const test = require('tap').test;
-
+const expect = require('expect');
 const gulp = require('gulp');
-const task = require('../');
+const { pipe } = require('mississippi');
 const path = require('path');
+const PluginError = require('plugin-error');
+const task = require('../');
 
 const filename = path.join(__dirname, './fixtures/pug-error.pug');
 
-test('should emit errors of pug correctly', function(t) {
-  gulp.src(filename)
-    .pipe(task()
-      .on('error', function(err) {
-        t.ok(err);
-        t.ok(err instanceof Error);
-        t.end();
-      }));
+describe('error', function() {
+  it('should emit errors of pug correctly', function(done) {
+    pipe(
+      [gulp.src(filename), task()],
```

Comment:
We should drop the dependency on gulp and use `miss.from.obj([file])` with a mocked file.

Reply:
You always need to end on a `miss.concat()`. If there's nothing to assert, no function needs to be passed.  The reason for this is that a gulpplugin is not a writeable stream and all streams must end with a writeable otherwise they will never be drained.

Reply:
In my tests, the stream was still drained. Still, I added `concat()` to be more explicit that the stream need be drained. A "sink" stream would be even better (drain everything without any processing).

Reply:
Fixed